ee/tables/execparsers/flatpak/remote_ls/upgradeable/parser.go

```diff
@@ -0,0 +1,42 @@
+package flatpak_upgradeable
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+)
+
+// Wow this has been a head-pounding bugger of a thing.
+//
+// flatpak remote-ls --updates
+// data is separated by whitespace only
+// data headers aren't in stdout, they are written directly to TTY
+// data column availability differs between linux distributions
+// data columns can only be specified on certain linux distributions with `--columns=`
+// columns are spaced arbitrarily to fit the window, both in flatpak *pretty* mode and simple mode. `export FLATPAK_FANCY_OUTPUT=0`
+// data will be ellipsed if it overflows the column's arbitrary width
+// // this happens to a higher degree if the command is ran with `script -c <cmd>` in an attempt to parse the printed headers (the column widths are based on window size from ioctl_tty)
+// Take a look at this beauty if you want to: [flatpak-table-printer.c](https://github.com/flatpak/flatpak/blob/main/app/flatpak-table-printer.c)
+//
+// After traveling various paths of acquiring and preserving as much data as I could, I give up. You win flatpak.
+// Thus is why I've resorted to good ole regexp, and only to get the app id.
+// The app id is the only* consistent data output between linux distributions, and is used in the flatpak update command.
+// The app id conforms to: [dbus specification](https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names)
+func remotelsParse(reader io.Reader) (any, error) {
+	results := make([]map[string]string, 0)
+	re := regexp.MustCompile("(([a-zA-Z0-9_]+\\.){1,}([a-zA-Z0-9_]+))")
```

Comment:
Since `MustCompile` will panic if given an invalid regex, we usually use it for defining global variables and not within functions (e.g. like [this](https://github.com/kolide/launcher/blob/main/cmd/launcher/uninstall.go#L15)) -- could you move this above so we can get similar safety?

(Optional -- if you use backticks instead of double quotes, you should be able to avoid the extra escape for the period -- `(([a-zA-Z0-9_]+\.){1,}([a-zA-Z0-9_]+))`. I like this because it's easier to paste into regex101 🙂 but not a requirement.)

Reply:
Honestly I usually use backticks and am not sure why I didn't try this time 😅